signer/services.go

```diff
@@ -1,17 +1,65 @@
 package signer
 
 import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
 	"sync"
 
 	tmLog "github.com/tendermint/tendermint/libs/log"
 	tmOS "github.com/tendermint/tendermint/libs/os"
 	tmService "github.com/tendermint/tendermint/libs/service"
 )
 
-func WaitAndTerminate(logger tmLog.Logger, services []tmService.Service) {
+func RequireNotRunning(lockFilePath string) error {
+	if _, err := os.Stat(lockFilePath); err != nil {
+		if os.IsNotExist(err) {
+			// lock file does not exist, can continue starting daemon
+			// or performing other tasks that require horcrux daemon to be stopped.
+			return nil
+		}
+		return fmt.Errorf("unexpected error while checking for existence of lock file at %s: %w", lockFilePath, err)
+	}
+
+	lockFile, err := os.ReadFile(lockFilePath)
+	if err != nil {
+		return fmt.Errorf("error reading lock file: %s, %w", lockFilePath, err)
+	}
+
+	trimmed := strings.TrimSpace(string(lockFile))
+
+	pid, err := strconv.ParseInt(trimmed, 10, 64)
+	if err != nil {
+		return fmt.Errorf("unexpected error parsing PID from lock file: %s. manual deletion of lockfile required. %w",
+			lockFilePath, err)
+	}
+
+	_, err = os.FindProcess(int(pid))
```

Comment:
> On Unix systems, FindProcess always succeeds and returns a Process for the given pid, regardless of whether the process exists.

Looks like the correct approach is to send it signal 0 to see if the process is alive: https://unix.stackexchange.com/a/169899

Reply:
From what I was reading, it won't have permission to do that unless it was a child process, but I'll try it out

Reply:
It worked!